### ISSUE TITLE
Some pods do not start after delete and install of UERANSIM

### ISSUE THREAD

**Issue by pinoOgni** [Orange-OpenSource/towards5gs-helm#55]
Hi folks, I hope you can help me solve this problem. 

I have a cluster created with kubeadm with 2 physical nodes called cube2 and cube4. 
More info:
* Ubuntu 18.04.5 LTS
* Kernel version  5.8.5-050805-generic
* Container Runtime version: containerd://1.5.5
* Kubectl Version: v1.23.3
* K8s Version: v1.23.8
* Calico version: v3.23.2 (installed with IP forwarding)
* Multus CNI version: 3.9
* The nodes interfaces are in promiscuous mode and their name is e0 (with the altname property)

Everything works perfectly, both the free5gc and ueransim pods are correctly deployed, the uesimtun0 interface is created and I can contact the internet from the UE. Here a screeshot:

![work](https://user-images.githubusercontent.com/26680903/191937241-896df4ae-4549-49f6-a33a-b2de907fc018.png)

Now before exposing the problem(s) I would like to point out that right now I am using a local version of the repo that refers to this [commit](https://github.com/Orange-OpenSource/towards5gs-helm/commit/cb033169c62883fc9e1ca980a25646579021947a) because in that version everything worked fine (some weeks ago) and I was thinking of solving the problem using that version but I was wrong.

The first problem occurs when I run the following commands:
```
helm delete -n 5g ueransim
# I wait for the pods to be terminated and then
helm -n 5g install ueransim ./towards5gs-helm/charts/ueransim/ --set global.n2network.masterIf=e0,global.n3network.masterIf=e0
```
After some time, this is the situation:
![not_work_first](https://user-images.githubusercontent.com/26680903/191935148-01bda171-897d-489c-bd23-43b409754079.png)

Note that the PODs in unknown state are amf, smf and upf and the gnb is in pending. The only thing these PODs have in common is that they use MACVLANs and have multiple interfaces configured with Multus.

Since the previous situation did not change, I ran these commands:
```
helm delete -n 5g ueransim
helm delete -n 5g free5gc
# I wait for the pods to be terminated and then
helm -n 5g install free5gc ./towards5gs-helm/charts/free5gc --set global.n2network.masterIf=e0,global.n3network.masterIf=e0,global.n4network.masterIf=e0,global.n6network.masterIf=e0,global.n9network.masterIf=e0,global.n6network.subnetIP=<subnetIP>,global.n6network.cidr=<cidr>,global.n6network.gatewayIP=<gatewayIP>,free5gc-upf.upf.n6if.ipAddress=<fakeAdressIP>
helm -n 5g install ueransim ./towards5gs-helm/charts/ueransim/ --set global.n2network.masterIf=e0,global.n3network.masterIf=e0
```

After some time, this is the situation:

![not_work_end](https://user-images.githubusercontent.com/26680903/191936118-27e5d460-b3c9-4852-b322-061647465b41.png)

Note that the status of the amf, smf and upf pods was the same even before the UERANSIM installation.

I also upload the "kubectl describe pod " of upf. The same situation can be seen from the logs of: amf, smf and gnb. Again as before, the only thing these PODs have in common is that they use MACVLANs and have multiple interfaces configured with Multus.

![upf](https://user-images.githubusercontent.com/26680903/191936706-0e692fc6-ac82-4e18-877f-5eb75c7af39b.png)

Do you have any suggestions? Thanks in advance

EDIT: I'm trying to understand the error better and maybe it's related to Multus and MACVLAN plugin.

**Comment by raoufkh** [comment]
Hi

Thank you first for the clear explanation.

Can you share the result of this command `kubectl -n <your-namespace> get network-attachment-definitions` at each step (i.e. 1) after installing Free5GC and UERANSIM 2) after deleting UERANSIM 3) after reinstalling UERANSIM, etc.

Raouf

**Comment by pinoOgni** [comment]
Hi Raouf thanks for you response, here it is:

**Comment by pinoOgni** [comment]
Hi Raouf, thanks for the response, here it is:

1. after installing Free5GC and UERANSIM
![free5gc_and_ueransim](https://user-images.githubusercontent.com/26680903/197505486-4927714a-5415-43ef-9102-b829cc6c22d8.png)

2. after deleting UERANSIM
![ueransim_deleted](https://user-images.githubusercontent.com/26680903/197505538-b84855c9-2de0-419a-bac2-bdb2bf549c95.png)

3. after reinstalling UERANSIM 
![ueransim_installed_wait_time](https://user-images.githubusercontent.com/26680903/197506637-2ec9782e-7502-4f31-a064-84f3eeeac0f9.png)

I have also noticed that the following pods go into Unknown state but then restart and continue to function normally.

![other_pods](https://user-images.githubusercontent.com/26680903/197506956-9ee4de93-1b84-4908-bad3-2226fb9cb24e.png)
 
4. after deleting UERANSIM and Free5GC
![free5gc_ueransim_reinstalled](https://user-images.githubusercontent.com/26680903/197507011-1023ccb9-b1e8-4c7f-bc24-ec60499c8f50.png)


Anyway, immediately after writing this issue, I had found a trick to overcome the problem. In practice since this sequence of steps does not work:
```
helm delete -n 5g ueransim
helm delete -n 5g free5gc
# I wait for the pods to be terminated and then
helm -n 5g install free5gc ./towards5gs-helm/charts/free5gc --set global.n2network.masterIf=e0,global.n3network.masterIf=e0,global.n4network.masterIf=e0,global.n6network.masterIf=e0,global.n9network.masterIf=e0,global.n6network.subnetIP=<subnetIP>,global.n6network.cidr=<cidr>,global.n6network.gatewayIP=<gatewayIP>,free5gc-upf.upf.n6if.ipAddress=<fakeAdressIP>
helm -n 5g install ueransim ./towards5gs-helm/charts/ueransim/ --set global.n2network.masterIf=e0,global.n3network.masterIf=e0
```

Just reverse the cancellation order like this:

```
helm delete -n 5g free5gc
helm delete -n 5g ueransim
# I wait for the pods to be terminated and then
helm -n 5g install free5gc ./towards5gs-helm/charts/free5gc --set global.n2network.masterIf=e0,global.n3network.masterIf=e0,global.n4network.masterIf=e0,global.n6network.masterIf=e0,global.n9network.masterIf=e0,global.n6network.subnetIP=<subnetIP>,global.n6network.cidr=<cidr>,global.n6network.gatewayIP=<gatewayIP>,free5gc-upf.upf.n6if.ipAddress=<fakeAdressIP>
helm -n 5g install ueransim ./towards5gs-helm/charts/ueransim/ --set global.n2network.masterIf=e0,global.n3network.masterIf=e0

```

Giuseppe

**Comment by raoufkh** [comment]
I think there is a problem of Calico and Multus working together in your cluster. There are problems in the management of additional interfaces. I think it would be useful to publish an issue on the rest of calico and multus to find a more sophisticated solution because with flannel+multus I can uninstall and reinstall without problems.

Raouf

**Comment by pinoOgni** [comment]
Yes I think you are right. I'll do it, thanks anyway for the help.

**Comment by raoufkh** [comment]
Hello @pinoOgni 

Did you manage to make it work? 

Bests
Abderaouf

**Comment by pinoOgni** [comment]
Hi @raoufkh  not yet, the only trick I could find is what I wrote. I still think it's related to the macvlan cni plugin which is downloaded when you install the container runtime (in my case containerd). I don't know if it depends on the version installed or on some configuration between the CNI used and Multus.


**Note** that I don't know if it's related: I've had a similar error happen to me in some cases. When I install the Core, the amf, smf pods stay in the Init phase, while the upf is running but if I try to contact it from another pod it doesn't work and also it has no internet connection (stupidly i can't do `apt update` from inside the upf). 
The only thing they have in common is the MACVLAN and as a simple test I deleted the cni plugin in `/opt/cni/bin` of that node and put that of a node where I didn't have this problem and magically after a new installation all pods started working. I'll try to investigate a little better if I can.
